.github/workflows/testing-and-deployment.yml

```diff
@@ -53,7 +53,14 @@ jobs:
         run: pytest -v --ignore=tests/plotting
 
       - name: Test Plotting
-        run: pytest -v tests/plotting --fail_extra_image_cache
+        run: pytest -v tests/plotting --fail_extra_image_cache --generated_image_dir _debug_images
+
+      - name: Upload Images for Failed Tests
+        if: ${{ failure() && hashFiles('_debug_images/*.png') != '' }}
```

Comment:
What is the reason for the hashFiles check?  I assume to not upload when no files are present?  Why not just upload an empty directory in this case?  That would be my preference.

Reply:
Yes it's for no upload when empty. This was left over from the doc image testing. For that test there is a specific dir created for failed tests only. The plotting tests are different, and all images (failed or not) are generated. So yea we can always upload this

Reply:
Resolved in https://github.com/pyvista/pyvista/pull/6204/commits/fd251e7f520544ca24b8123df71eb5f498f34245